packages/react-native/src/posthog-rn.ts

```diff
@@ -68,6 +68,13 @@ export type PostHogOptions = PostHogCoreOptions & {
    * Defaults to false
    */
   enablePersistSessionIdAcrossRestart?: boolean
+
+  /**
+   * A list of headers that should be sent with requests to the PostHog API.
+   *
+   * @default {}
+   */
+  request_headers?: { [header_name: string]: string }
```

Comment:
lets add this to `PostHogCoreOptions` so the other packages will benefit from this too

Reply:
We should add a note that this won't work for react native - session replay since the native SDKs (Android and iOS SDKs) send the recordings, so the headers would need to be set there as well.
So this feature has to be added there first